main.js

```diff
@@ -28,13 +28,14 @@ class App extends React.Component {
     username: undefined,
   };
   componentDidMount() {
-    Linking.addEventListener('url', this._handleAuth0Redirect);
+    Linking.addEventListener('url', this._handleAuth0Redirect.bind(this));
   }
 
   _loginWithAuth0 = async () => {
     const redirectionURL = `${auth0Domain}/authorize` + this._toQueryString({
       client_id: auth0ClientId,
-      response_type: 'token',
+      response_type: 'id_token',
+      nonce: 'alongrandomstringtopreventtokenreplayattacks',
```

Comment:
Maybe this should be handled by exactly random string? :) [Auth0 example](https://auth0.com/docs/api-auth/tutorials/nonce) seems to be handling this in nicer way :)

Reply:
I wanted to keep this PR simple, but yeah a random string store into asyncStorage is better for production requirements ;) So as you want, I can add this security part or not